contrib/database/sql/exec_trace_test.go

```diff
@@ -3,14 +3,6 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2023 Datadog, Inc.
 
-// TODO: gotraceui does not currently handle Go 1.21 execution tracer changes,
-// so we need to skip this test for that version. We still have coverage for
-// older Go versions due to our support policy, and Go 1.21 shouldn't fundamentally
-// change the behavior this test is covering. Remove this build constraint
-// once gotraceui supports Go 1.21
-//
-//go:build !go1.21
```

Comment:
We can't remove this guard completely. We at least need to guard against Go 1.22, which introduces changes to the execution trace format that gotraceui doesn't support yet.

Reply:
@nsrip-dd Updated. Please check if it feels right with the new wording.

Reply:
LGTM, thanks!

Reply:
I see you reverted back to the go1.21 tag, was something not working? If it's too much trouble it might be better to postpone this upgrade. One of the Go runtime maintainers is working on a new package which should cover the old and new execution tracer formats. I might just switch to that once it's ready to use.

Reply:
@nsrip-dd I'm testing something really strange. Reviewdog fails to run in this PR ([example](https://github.com/DataDog/dd-trace-go/actions/runs/7264716396/job/19792843567)). I kind of reproduced locally running the lint job, but I have no simple way to debug.

And I'm going to revert the merge. I forgot automerge was activated.